playwright/tests/common/environment.spec.ts

```diff
@@ -4,6 +4,31 @@ import { test } from "../../fixtures";
 import { mockManagerProcedures } from "../../utils";
 import { environments } from "../../mocks";
 
+function buildMockEnvironmentProcedures(args: {
```

Comment:
Personally, I think wouldn't have done that, it creates a bit more complexity when reading tests. I had to check above when you slice or take only one, which one. What does it mean, etc.

I feel the repetition super light here, no? If you just have a simple array most of the time but yes you repeat the path, I think it's easier to maintain and to read. 

To maintain shared small function like this, we can break all tests while just wanting to update one. I let you see what you think if it's flatten and judge it. I just want us to try to have less overhead when reading tests, even if it means a bit of repetition. And since any failure / problem will actually make the test fail it's easy to maintain.

Reply:
I agree; I'm not a big fan of builder functions like this, especially because eventually that function will grow to support more configuration rather than inlining the contents where it is used.

The repetition comes from the `fetchEnvironments` mock. It needs to be repeated every time we call `mockManagerProcedures`; only the most recent `mockManagerProcedures` call's procedures seemed to be registered.

Because the environment switching logic is reused in several tests, the file became long and repetitive.

If we can get it to work, this is a nicer solution in my opinion: https://github.com/prismicio/slice-machine/pull/1253#discussion_r1434510174